python-checks/src/test/java/org/sonar/python/checks/ImplicitStringConcatenationCheckTest.java

```diff
@@ -43,61 +45,91 @@ public void simple_expression_quickfix() {
+      "");
+    String codeFixed2 = code(
+      "def a():",
+      "    b = ['1' + '2']",
+      "");
     PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed1, codeFixed2);
   }
 
   @Test
   public void concat_expression() {
-    String codeWithIssue = "def a():\n" +
-      "    b = ['1'+'2' '3'+'4']\n";
-    String codeFixed1 = "def a():\n" +
-      "    b = ['1'+'2', '3'+'4']\n";
-    String codeFixed2 = "def a():\n" +
-      "    b = ['1'+'2' + '3'+'4']\n";
+    String codeWithIssue = "['1'+'2' '3'+'4']";
+    String codeFixed1 = "['1'+'2', '3'+'4']";
+    String codeFixed2 = "['1'+'2' + '3'+'4']";
     PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed1, codeFixed2);
   }
 
   @Test
   public void sets() {
-    String codeWithIssue = "def a():\n" +
-      "    {'1' '2', '3'}\n";
-    String codeFixed1 = "def a():\n" +
-      "    {'1', '2', '3'}\n";
-    String codeFixed2 = "def a():\n" +
-      "    {'1' + '2', '3'}\n";
+    String codeWithIssue = "{'1' '2', '3'}";
+    String codeFixed1 = "{'1', '2', '3'}";
+    String codeFixed2 = "{'1' + '2', '3'}";
     PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed1, codeFixed2);
   }
 
   @Test
   public void parameters() {
-    String codeWithIssue = "def a():\n" +
-      "    print('1' '2', '3')\n";
-    String codeFixed1 = "def a():\n" +
-      "    print('1', '2', '3')\n";
-    String codeFixed2 = "def a():\n" +
-      "    print('1' + '2', '3')\n";
+    String codeWithIssue = "print('1' '2', '3')";
+    String codeFixed1 = "print('1', '2', '3')";
+    String codeFixed2 = "print('1' + '2', '3')";
     PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed1, codeFixed2);
   }
 
   @Test
   public void parameters2() {
-    String codeWithIssue = "def a():\n" +
-      "    foo('1' '2', '3')\n";
-    String codeFixed1 = "def a():\n" +
-      "    foo('1', '2', '3')\n";
-    String codeFixed2 = "def a():\n" +
-      "    foo('1' + '2', '3')\n";
+    String codeWithIssue = "foo('1' '2', '3')";
+    String codeFixed1 = "foo('1', '2', '3')";
+    String codeFixed2 = "foo('1' + '2', '3')";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed1, codeFixed2);
+  }
+
+  @Test
+  public void multiline_list() {
+    String codeWithIssue = code("a = [",
+      "  '1'",
+      "  '2'",
+      "]");
+    String codeFixed1 = code("a = [",
+      "  '1',",
+      "  '2'",
+      "]");
+    String codeFixed2 = code("a = [",
+      "  '1' + '2'",
+      "]");
     PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed1, codeFixed2);
   }
 
+  @Test
+  public void multiline_list_with_comment() {
+    String codeWithIssue = code("a = [",
+      "  '1' # comment",
+      "  '2'",
+      "]");
+    String codeFixed1 = code("a = [",
+      "  '1', # comment",
+      "  '2'",
+      "]");
+    String codeFixed2 = code("a = [",
+      "  '1' + '2'",
+      "]");
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed1, codeFixed2);
+  }
```

Comment:
What in case of the formatting from the ticket (more parameters, multi-line):

```
def my_list_with_comment():
    x = ["a",
         "b",
         "c",
         "d"  # A comment
         "e", # Another comment
         "f"]
    print(x)
```

I suggest adding it to test cases if it's not covered yet